recipe-box-backend/src/main/java/com/endava/recipebox/controller/RecipeController.java

```diff
@@ -39,8 +40,13 @@ public ResponseEntity<List<RecipeDTO>> getAllPublicRecipesByName(@RequestParam S
     }
 
     @PostMapping
-    public ResponseEntity<String> createRecipe(@Validated @RequestBody RecipeAddRequestDTO recipeAddRequestDTO, @RequestParam Long userId) {
+    public ResponseEntity<?> createRecipe(@Validated @RequestBody RecipeAddRequestDTO recipeAddRequestDTO,@RequestParam Long userId) {
         return ResponseEntity.ok(recipeService.createRecipe(recipeAddRequestDTO, userId));
     }
 
+    @PatchMapping
```

Comment:
Use PUT: When you need to completely replace a resource or when the client can send the entire resource representation.
Use PATCH: When you only need to update specific fields of a resource without modifying the entire resource.
My recommandation is tu use Patch because from frontend will be send the entire body and not only the fields updated.